django/university/exchange/utils.py

```diff
@@ -156,3 +160,42 @@ def convert_sigarra_schedule(schedule_data):
         new_schedule_data.append(new_schedule)
 
     return new_schedule_data
+
+def update_schedule_accepted_exchanges(student, schedule, cookies):
+    direct_exchange_ids = DirectExchangeParticipants.objects.filter(
+        participant=student, direct_exchange__accepted=True
+    ).values_list('direct_exchange', flat=True)
+    direct_exchanges = DirectExchange.objects.filter(id__in=direct_exchange_ids)
+
+    for exchange in direct_exchanges:
+        participants = DirectExchangeParticipants.objects.filter(direct_exchange=exchange, participant=student)
+        (status, trailing) = update_schedule(schedule, participants, cookies) 
+        if status == ExchangeStatus.FETCH_SCHEDULE_ERROR:
+            return (ExchangeStatus.FETCH_SCHEDULE_ERROR, trailing)
+
+    return (ExchangeStatus.SUCCESS, None)
+
+def update_schedule(student_schedule, exchanges, cookies):
+    (semana_ini, semana_fim) = curr_semester_weeks();
+
+    for exchange in exchanges:
+        for i, schedule in enumerate(student_schedule):
+            if schedule["ucurr_sigla"] == exchange.course_unit:
+                ocorr_id = schedule["ocorrencia_id"]
+                class_type = schedule["tipo"]
+
+                unit_schedules = requests.get(get_unit_schedule_url(
+                    ocorr_id,
+                    semana_ini,
+                    semana_fim
+                ), cookies=cookies)
+
+                if unit_schedules.status_code != 200:
+                    return (ExchangeStatus.FETCH_SCHEDULE_ERROR, unit_schedules.status_code)
+
+                for unit_schedule in unit_schedules.json()["horario"]:
+                    for turma in unit_schedule["turmas"]:
+                        if turma["turma_sigla"] == exchange.new_class and unit_schedule["tipo"] == class_type:
+                            student_schedule[i] = unit_schedule
```

Comment:
If you'd like, please write a comment explaining that we are not deleting the old class key because it doesn't matter when returning the data from the backend